flink-clients/src/main/java/org/apache/flink/client/deployment/AbstractContainerizedClusterClientFactory.java

```diff
@@ -36,9 +38,10 @@
 	public ClusterSpecification getClusterSpecification(Configuration configuration) {
 		checkNotNull(configuration);
 
-		final int jobManagerMemoryMB = ConfigurationUtils
-			.getJobManagerHeapMemory(configuration)
-			.getMebiBytes();
+		final JobManagerProcessSpec jobManagerProcessSpec = JobManagerProcessUtils.processSpecFromConfig(
+			JobManagerProcessUtils.getConfigurationWithLegacyHeapSizeMappedToNewConfigOption(
```

Comment:
Unrelated:
I was tracking where this method `getConfigurationWithLegacyHeapSizeMappedToNewConfigOption` lead to, and find that `MemoryBackwardsCompatibilityUtils#getLegacyTaskManagerHeapMemoryIfExplicitlyConfigured` is named "TaskManager" but actually used for both JM & TM.